### ISSUE TITLE
Why can't I reproduce mnist_cnn.py in the Keras example?

### ISSUE THREAD

**Issue by soloice** [fchollet/keras#5010]
I'm using Tensorflow backend and a Quadro K600 GPU. My OS is win7 and Python 3.5 environment is Anaconda. When I run the mnist_cnn.py example code, I can only get an accuracy of 98.95% rather than 99.25% claimed in the code comment.

Is that normal?

This is my running log, not sure if this would help:
```
C:\Anaconda2\envs\py3\python.exe E:/WorkSpace/Python/tf-road/src/mnist_cnn.py
Using TensorFlow backend.
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:128] successfully opened CUDA library cublas64_80.dll locally
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:128] successfully opened CUDA library cudnn64_5.dll locally
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:128] successfully opened CUDA library cufft64_80.dll locally
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:128] successfully opened CUDA library nvcuda.dll locally
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\stream_executor\dso_loader.cc:128] successfully opened CUDA library curand64_80.dll locally
X_train shape: (60000, 28, 28, 1)
60000 train samples
10000 test samples
Train on 60000 samples, validate on 10000 samples
Epoch 1/12
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\core\common_runtime\gpu\gpu_device.cc:885] Found device 0 with properties: 
name: Quadro K600
major: 3 minor: 0 memoryClockRate (GHz) 0.8755
pciBusID 0000:02:00.0
Total memory: 1.00GiB
Free memory: 784.91MiB
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\core\common_runtime\gpu\gpu_device.cc:906] DMA: 0 
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\core\common_runtime\gpu\gpu_device.cc:916] 0:   Y 
I c:\tf_jenkins\home\workspace\release-win\device\gpu\os\windows\tensorflow\core\common_runtime\gpu\gpu_device.cc:975] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Quadro K600, pci bus id: 0000:02:00.0)
60000/60000 [==============================] - 71s - loss: 0.3902 - acc: 0.8782 - val_loss: 0.0889 - val_acc: 0.9718
Epoch 2/12
60000/60000 [==============================] - 71s - loss: 0.1377 - acc: 0.9591 - val_loss: 0.0584 - val_acc: 0.9813
Epoch 3/12
60000/60000 [==============================] - 66s - loss: 0.1051 - acc: 0.9687 - val_loss: 0.0496 - val_acc: 0.9836
Epoch 4/12
60000/60000 [==============================] - 66s - loss: 0.0880 - acc: 0.9738 - val_loss: 0.0422 - val_acc: 0.9862
Epoch 5/12
60000/60000 [==============================] - 66s - loss: 0.0782 - acc: 0.9771 - val_loss: 0.0392 - val_acc: 0.9877
Epoch 6/12
60000/60000 [==============================] - 65s - loss: 0.0706 - acc: 0.9797 - val_loss: 0.0359 - val_acc: 0.9878
Epoch 7/12
60000/60000 [==============================] - 66s - loss: 0.0636 - acc: 0.9813 - val_loss: 0.0372 - val_acc: 0.9872
Epoch 8/12
60000/60000 [==============================] - 65s - loss: 0.0591 - acc: 0.9828 - val_loss: 0.0331 - val_acc: 0.9882
Epoch 9/12
60000/60000 [==============================] - 66s - loss: 0.0570 - acc: 0.9832 - val_loss: 0.0311 - val_acc: 0.9894
Epoch 10/12
60000/60000 [==============================] - 66s - loss: 0.0517 - acc: 0.9849 - val_loss: 0.0308 - val_acc: 0.9898
Epoch 11/12
60000/60000 [==============================] - 66s - loss: 0.0510 - acc: 0.9849 - val_loss: 0.0306 - val_acc: 0.9898
Epoch 12/12
60000/60000 [==============================] - 66s - loss: 0.0482 - acc: 0.9859 - val_loss: 0.0318 - val_acc: 0.9895
Test score: 0.0318144207031
Test accuracy: 0.9895

Process finished with exit code 0
```

**Comment by rishikksh20** [comment]
Yes it's normal as I see the difference is not so huge. Actually accuracy of a neural network or any computational model not only depends on its hyper-parameters or random.seed , but it also depends on  computing environment like hardware (GPUs or CPUs) or version of used software(python and tensorflow)
etc.

**Comment by soloice** [comment]
Oh, thanks for your patient explanation! It's very helpful!

**Comment by keunwoochoi** [comment]
@soloice Please close the issue.
